File: VB/WinWebSolution.Module/Module.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports System.ComponentModel
Imports DevExpress.Persistent.Base
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Updating

Namespace WinWebSolution.Module
	Public NotInheritable Partial Class WinWebSolutionModule
		Inherits ModuleBase
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Overrides Sub ExtendModelInterfaces(ByVal extenders As ModelInterfaceExtenders)
			MyBase.ExtendModelInterfaces(extenders)
			extenders.Add(Of IModelCommonMemberViewItem, IModelListViewColumnOptions)()
		End Sub
	End Class
	Public Interface IModelListViewColumnOptions
		<Category("Behavior"), DefaultValue(True)> _
		Property AllowSort() As Boolean
		<Category("Behavior"), DefaultValue(True)> _
		Property AllowGroup() As Boolean
	End Interface
	<AttributeUsage(AttributeTargets.Field Or AttributeTargets.Property, Inherited := True, AllowMultiple := False)> _
	Public NotInheritable Class ListViewColumnOptionsAttribute
		Inherits ModelExportedValuesAttribute
		Public Sub New(ByVal allowSort As Boolean, ByVal allowGroup As Boolean)
			Me.AllowSort = allowSort
			Me.AllowGroup = allowGroup
		End Sub
		Private privateAllowSort As Boolean
		Public Property AllowSort() As Boolean
			Get
				Return privateAllowSort
			End Get
			Set(ByVal value As Boolean)
				privateAllowSort = value
			End Set
		End Property
		Private privateAllowGroup As Boolean
		Public Property AllowGroup() As Boolean
			Get
				Return privateAllowGroup
			End Get
			Set(ByVal value As Boolean)
				privateAllowGroup = value
			End Set
		End Property
		Public Overrides Sub FillValues(ByVal values As Dictionary(Of String, Object))
			values.Add("AllowSort", AllowSort)
			values.Add("AllowGroup", AllowGroup)
		End Sub
	End Class
End Namespace

File: CS/WinWebSolution.Module/Module.cs
using System;
using DevExpress.ExpressApp;
using System.ComponentModel;
using DevExpress.Persistent.Base;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Updating;

namespace WinWebSolution.Module {
    public sealed partial class WinWebSolutionModule : ModuleBase {
        public WinWebSolutionModule() {
            InitializeComponent();
        }
        public override void ExtendModelInterfaces(ModelInterfaceExtenders extenders) {
            base.ExtendModelInterfaces(extenders);
            extenders.Add<IModelCommonMemberViewItem, IModelListViewColumnOptions>();
        }
    }
    public interface IModelListViewColumnOptions {
        [Category("Behavior"), DefaultValue(true)]
        bool AllowSort { get; set; }
        [Category("Behavior"), DefaultValue(true)]
        bool AllowGroup { get; set; }
    }
    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, Inherited = true, AllowMultiple = false)]
    public sealed class ListViewColumnOptionsAttribute : ModelExportedValuesAttribute {
        public ListViewColumnOptionsAttribute(bool allowSort, bool allowGroup) {
            this.AllowSort = allowSort;
            this.AllowGroup = allowGroup;
        }
        public bool AllowSort { get; set; }
        public bool AllowGroup { get; set; }
        public override void FillValues(Dictionary<string, object> values) {
            values.Add("AllowSort", AllowSort);
            values.Add("AllowGroup", AllowGroup);
        }
    }
}